tests-arquillian/src/test/java/org/jboss/weld/tests/extensions/staticObserver/ExtensionWithStaticLifecycleEventObserver.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2017, Red Hat, Inc., and individual contributors
+ * by the @authors tag. See the copyright.txt in the distribution for a
+ * full listing of individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.weld.tests.extensions.staticObserver;
+
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.spi.Extension;
+import javax.enterprise.inject.spi.ProcessAnnotatedType;
+
+/**
+ * @author Antoine Sabot-Durand
+ */
+public class ExtensionWithStaticLifecycleEventObserver implements Extension {
+
+    static void processAnnotatedType(@Observes ProcessAnnotatedType<?> pat) {
+        pat.veto();
```

Comment:
Please remove this invocation. Right now, the test would pass even without your `ExtensionBeanDeployer` modification. If you remove the invocation, the test would fail without the modification.

Otherwise looks good now.